kernel/src/sched.rs

```diff
@@ -454,49 +454,45 @@ impl Kernel {
-                match scheduler.do_kernel_work_now(chip) {
-                    true => {
-                        // Execute kernel work. This includes handling
-                        // interrupts and is how code in the chips/ and capsules
-                        // crates is able to execute.
-                        scheduler.execute_kernel_work(chip);
-                    }
-                    false => {
-                        // No kernel work ready, so ask scheduler for a process.
-                        match scheduler.next(self) {
-                            SchedulingDecision::RunProcess((appid, timeslice_us)) => {
-                                self.process_map_or((), appid, |process| {
-                                    let (reason, time_executed) = self.do_process(
-                                        platform,
-                                        chip,
-                                        scheduler,
-                                        process,
-                                        ipc,
-                                        timeslice_us,
-                                    );
-                                    scheduler.result(reason, time_executed);
-                                });
-                            }
-                            SchedulingDecision::TrySleep => {
+        no_sleep: bool,
+    ) {
+        chip.watchdog().tickle();
+        unsafe {
+            // Ask the scheduler if we should do tasks inside of the kernel,
+            // such as handle interrupts. A scheduler may want to prioritize
+            // processes instead, or there may be no kernel work to do.
+            match scheduler.do_kernel_work_now(chip) {
+                true => {
+                    // Execute kernel work. This includes handling
+                    // interrupts and is how code in the chips/ and capsules
+                    // crates is able to execute.
+                    scheduler.execute_kernel_work(chip);
+                }
+                false => {
+                    // No kernel work ready, so ask scheduler for a process.
+                    match scheduler.next(self) {
+                        SchedulingDecision::RunProcess((appid, timeslice_us)) => {
+                            self.process_map_or((), appid, |process| {
+                                let (reason, time_executed) = self.do_process(
+                                    platform,
+                                    chip,
+                                    scheduler,
+                                    process,
+                                    ipc,
+                                    timeslice_us,
+                                );
+                                scheduler.result(reason, time_executed);
+                            });
+                        }
+                        SchedulingDecision::TrySleep => {
+                            if !no_sleep {
```

Comment:
```suggestion
                            // For testing, it may be helpful to
                            // disable sleeping the chip in case
                            // the running test does not generate
                            // any interrupts.
                            if !no_sleep {
```